R/stan_module.R

```diff
@@ -117,6 +124,16 @@ setMethod(
     }
 )
 
+#' model_prep. Populates the model section of a StanModule
+#' @param x A StanModule object
+#' @export
+model_prep <- function(x) {
+    if (length(x@priors) > 0) x@priors <- as.list(paste(names(x@priors), "~", x@priors))
+    x@model <- paste(h_bracket(x@priors), x@model)
+    x
+}
```

Comment:
I think this is fine but can we include a unit test just to make sure we aren't modifying the original object i.e. if we did
```
y <- model_prep(x)
```

Making sure that `x` is unchanged. I am pretty sure this is how S4 works as I don't think it modifies in place but I just want to be sure.

Reply:
Let me know if I understood this correctly. Later on, when I invoke this function inside the `as.character`, I create a new object `y` instead of overwriting `x`. 

https://github.com/Genentech/jmpost/blob/bb2ddf7349021cfc308fb64f7121a15b45c7e3fb/R/stan_module.R#L146-L150

Reply:
I just did some manual tests to confirm S4 doesn't  create references so yer this is all fine